### PR TITLE
feat: connect to postgres instance unit test

### LINKED ISSUE #27: feat: connect to postgres instance
Via @paulushcgcj:
` @marcionemec-daitan to keep working on making the code connect to postgres and oracle and execute a simple select`

### PR DESCRIPTION
**Pull request by marcionemec-daitan**
# Description

Postgres connection test unit 

Fixes #27 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

### CODE REVIEW

File: sync/Dockerfile [hunk 1]
```diff
@@ -12,12 +12,12 @@ ARG PIP_NO_CACHE_DIR="off" \
     PYTHONUNBUFFERED="1"
 
 # Copy files and install requirements
-WORKDIR /app
+WORKDIR /sync
```
Comment (DerekRoberts) [inline sync/Dockerfile:15]: Please do not change the default Docker directory.  Leave it as `/app`.

File: sync/Dockerfile [hunk 2]
```diff
@@ -12,12 +12,12 @@ ARG PIP_NO_CACHE_DIR="off" \
...
 COPY requirements.txt .
 RUN pip install -r ./requirements.txt
 
 # Start the app
-WORKDIR /app/src
+WORKDIR /sync/src
```
Comment (DerekRoberts) [inline sync/Dockerfile:20]: Please do not change the default Docker directory. Leave it as /app.

File: sync/src/main.py [hunk 3]
```diff
@@ -5,34 +5,86 @@
...
+        print("Required variable tests passed!")
     else:
         raise Exception("Not all required variables to execute a instance of Data Sync Engine exists.")
     
+def testOracleConnection():
+    print("-- 3. Checking if Oracle connection is available and reachable")
+    dbConfig = db_config_test("ORACLE","THE") 
+    response = os.system("ping -c 1 " + dbConfig["host"] )
+    if response == 0:
+        print("ORACLE Host ("+dbConfig["host"]+") is reachable")
```
[the remark this answers is not in the document]
Comment (DerekRoberts) [reply sync/src/main.py:69]: Please either resolve this or explain why it's safe to ignore.

File: sync/src/main.py [hunk 4]
```diff
@@ -5,34 +5,86 @@
...
+    else:
+        print("ORACLE Host ("+dbConfig["host"]+") is unreachable")
```
[the remark this answers is not in the document]
Comment (DerekRoberts) [reply sync/src/main.py:71]: Please either resolve this or explain why it's safe to ignore.

File: sync/src/module/test_db_connection.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+            for record in results:
+                print(record)
+            return results
+    
+    def do_test_old(database_config):
+        """ Test Connection """
+        connection_string = test_db_connection.format_connection_string(database_config)
+        ssl_args = test_db_connection.get_ssl_args(database_config)
+        print(ssl_args)
+        print(connection_string)
```
[the remark this answers is not in the document]
Comment (DerekRoberts) [reply sync/src/module/test_db_connection.py:38]: Please either resolve this or explain why it's safe to ignore.

File: sync/src/module/test_db_connection.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+            ssl_args = {'ssl_context': ssl_context}
+            # ssl_args = {'ssl':{'fake_flag_to_enable_tls': True}}
+            
+        return ssl_args        
+        
+    def format_connection_string(database_config: str):
+        """ Formats the connection string based on the database type and the connection configuration. """
+        if database_config['type'] == 'ORACLE':
+            import oracledb
+            print("Oracle DB Connection to " + database_config['host'] + ", SN=" + database_config['service_name'])
```
[the remark this answers is not in the document]
Comment (DerekRoberts) [reply sync/src/module/test_db_connection.py:60]: Please either resolve this or explain why it's safe to ignore.

**Review by DerekRoberts — CHANGES_REQUESTED** [review]

File: sync/src/main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcionemec-daitan) [reply sync/src/main.py:69]: This is a test to verify if the DB host is reachable. No security threat, IMO. Anyway, I will remove it, as it is still unnecessary as I'm not trying to connect in Oracle DB yet.

File: sync/src/main.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (marcionemec-daitan) [reply sync/src/main.py:71]: Same as above

File: sync/src/module/test_db_connection.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (marcionemec-daitan) [reply sync/src/module/test_db_connection.py:38]: I will remove this unnecessary print as it is not needed to be tested each time.

File: sync/src/module/test_db_connection.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (marcionemec-daitan) [reply sync/src/module/test_db_connection.py:60]: Same as above

File: sync/Dockerfile (hunk 1 above)
Comment (marcionemec-daitan) [reply sync/Dockerfile:15]: I will undo that change

**Pull request merged**